src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -246,6 +257,36 @@ public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, in
             }
         }
 
+        private bool TryNavigateToMappedFile(Workspace workspace, ISpanMappingService spanMappingService, Document generatedDocument, TextSpan textSpan)
+        {
+            if (workspace is VisualStudioWorkspaceImpl vsWorkspace)
```

Comment:
when would then ever not be the case if we have a mapped doc?

Reply:
It shouldn't be the case, but I have to cast it to access methods on it.  I was just being cautious here in case I was wrong - should I instead just directly cast it?

Reply:
it's fine to be cautious.  but we should also state/check our invariants.  i.e. if this was not a VSWorkspace, we should likely assert so that we can be made aware that something is awry.  this allows us to not crash if things go bad, but also get awareness instead of silence on the issue.